xla/service/gpu/gpu_windowed_einsum_handler.cc

```diff
@@ -567,6 +538,34 @@ bool ShouldAddToChain(const HloInstruction* inst) {
       return false;
   }
 }
+absl::Status PostProcessUnrolledLoop(HloInstruction* loop, int64_t stream_id) {
+  HloComputation* while_body = loop->while_body();
+  // This is to set force delay for the first collective permute so it can
+  // be scheduled always at the top of computation. The GTE index it's consuming
+  // is 2 for RS loop; 0 for AG loop.
+  int64_t force_delay_cp_gte_index =
+      while_body->name().find(
+          GpuWindowedEinsumHandler::kWindowedEinsumRsLoopName) == 0
```

Comment:
This seems fragile. Can we generalize this? What is the characteristic why you want to schedule it first? 

Why is the index 2 or 0? Does this rely on the previously generated IR?

Reply:
This is to enforce the order of comm-gemm and gemm+gemm so we get the best overlap, we manually control the order of instructions to be comm->gemm->comm->gemm.
The indices are fixed in the previously generated IR, when SPMD partitioner creates the CM loop, it sets these indices in the loop tuple.